org.elysium.parent/org.elysium.ui/src/org/elysium/ui/contentassist/LilyPondProposalProvider.java

```diff
@@ -41,33 +74,114 @@ public void completeVersion_Version(EObject model, Assignment assignment, Conten
 	@Override
 	public void completeInclude_ImportURI(EObject model, Assignment assignment, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
 		super.completeInclude_ImportURI(model, assignment, context, acceptor);
-		IResource resource = ResourceUtils.convertEResourceToPlatformResource(model.eResource());
+		IResource resource=getResource(model);
 		if (resource != null) {
-			List<String> paths = Lists.newArrayList();
-			paths.add(resource.getParent().getLocation().toOSString());
-			paths.add(LilyPondImportUriResolver.getDefaultSearchUri(lilyPondPathProvider.getLilyPondPath()).getPath());
-			paths.addAll(Lists.newArrayList(lilyPondPathProvider.getSearchPaths()));
+			String searchPrefix=getImportSearchPrefix(context);
+			boolean absolute=LilyPondImportUriResolver.isAbsolute(searchPrefix, LilyPondConstants.IS_WINDOWS);
+
+			List<String> paths = getSearchPaths(resource, absolute);
 
 			for (String path : paths) {
-				String[] children = new File(path).list(new FilenameFilter() {
+				java.net.URI pathToSearch = LilyPondImportUriResolver.saferResolve(new File(path).toURI(), searchPrefix, LilyPondConstants.IS_WINDOWS);
+				File folder = new File(pathToSearch).getParentFile();
+				if(folder.exists()) {
+					File[] allFiles = folder.listFiles();
+					for (File file : allFiles) {
+						if(propose(file)) {
+							String name=file.getName();
+							String proposal=mergePrefixAndProposedName(context.getPrefix().substring(1), name);
```

Comment:
If I invoke content assist directly after `\include` before typing `"`, the following exception is thrown here:
```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1931)
	at org.elysium.ui.contentassist.LilyPondProposalProvider.completeInclude_ImportURI(LilyPondProposalProvider.java:92)
```

Reply:
fixed including additional test case